shell/platform/linux/fl_key_event_channel.cc

```diff
@@ -0,0 +1,54 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_key_event_channel.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_json_message_codec.h"
+
+struct _FlKeyEventChannel {
+  FlBasicMessageChannel parent_instance;
+};
+
+G_DEFINE_TYPE(FlKeyEventChannel,
+              fl_key_event_channel,
+              fl_basic_message_channel_get_type())
+
+static void fl_key_event_channel_class_init(FlKeyEventChannelClass* klass) {}
+
+static void fl_key_event_channel_init(FlKeyEventChannel* self) {}
+
+FlKeyEventChannel* fl_key_event_channel_new(FlBinaryMessenger* messenger) {
+  g_return_val_if_fail(FL_IS_BINARY_MESSENGER(messenger), nullptr);
+
+  g_autoptr(FlJsonMessageCodec) codec = fl_json_message_codec_new();
+  FlKeyEventChannel* self = static_cast<FlKeyEventChannel*>(
+      g_object_new(fl_key_event_channel_get_type(), "messenger", messenger,
+                   "name", "flutter/keyevent", "codec", codec, nullptr));
+
+  return self;
+}
+
+void fl_key_event_channel_send_key_event(FlKeyEventChannel* channel,
+                                         const gchar* type,
+                                         int64_t key_code,
+                                         int64_t scan_code,
+                                         int64_t modifiers,
+                                         int64_t unicodeScalarValues) {
+  g_autoptr(FlValue) message = fl_value_new_map();
+  fl_value_set_string_take(message, "type", fl_value_new_string(type));
+  fl_value_set_string_take(message, "keymap", fl_value_new_string("linux"));
+  fl_value_set_string_take(message, "keyCode", fl_value_new_int(key_code));
+  fl_value_set_string_take(message, "scanCode", fl_value_new_int(scan_code));
+  fl_value_set_string_take(message, "modifiers", fl_value_new_int(modifiers));
+  // FIXME(robert-ancell) Using GLFW toolkit as Flutter doesn't understand
```

Comment:
See https://github.com/flutter/flutter/pull/54227 for what's involved for this.

Reply:
I'll do this in a follow up PR.